python_wrapper/pre-compile.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# the procedure for adding a new ext dependency to be bundled in here:
+# - add git checkout, compile, etc
+# - ensure the version ends up in python_wrapper/src/versions.txt
+# - ensure the licence ends up in python_wrapper/src/copying/, and fname is referenced in copying/list.json
+# - ensure the .so ends up in target/lib64/ with the expected libname
+# - validate that the resulting wheel contains all the above
+# additionally, make sure this script is aligned with /buildscripts/compile.sh and /buildscripts/wheel-linux.sh,
+# in particular when it comes to install targets and package data, etc
+
+# note also that for macos, we assume that the agent has the libraries already installed, as we can't run this in docker
+# we thus only prepare the license files
+
+set -euo pipefail
+
+mkdir -p python_wrapper/src/copying
+mkdir -p /tmp/eckit/target/eckit/lib64/
+
+if [ "$(uname)" != "Darwin" ] ; then
+    echo "installing deps for platform $(uname)"
+
+    ## curl -- handled by docker container being built with:
+    # dnf instal -y libpsl-devel openssl-devel curl-devel
+    ## this below if the build would need a customization:
+    # git clone https://github.com/curl/curl /src/curl
+    # cd /src/curl
+    # autoreconf -fi
+    # ./configure --prefix /tmp/curl --with-openssl
+    # make -j10 && make install
+    # cd -
+
+    ## proj
+    git clone https://github.com/OSGeo/PROJ /src/proj
+    mkdir /src/proj/build && cd /src/proj/build
+    cmake .. -DCMAKE_INSTALL_PREFIX=/tmp/proj
+    cmake --build . -j8
+    cmake --build . --target install
```

Comment:
@pmaciel I've added the `proj` build here. I've originally started with just `dnf install -y proj-devel`, but that fails to be picked later by ecbuild (and the `PROJ_INCLUDE_DIRS` etc don't seem to be exposed), so I just compile. It allows us to additionally specify whether we want to have eg `libtiff` dependency. And I've replaced the manual curl build with dnf install, that does seem to work ok

tested a linux docker build, seems to work ok. For macos I still need to test a build and possibly add some brew installs, I'll do that once I'll have a "working on a macos laptop" day

Reply:
I'm quite surprised that curl-devel works, but proj-devel not (proj has better CMake modules support.) So yes this solution works, but we need to revisit this soon to make sure it works as intended. It would be better to fix this now, before the whole workflow is complete (it's slow to improve on working solutions, even if they're costly to maintain.)

There's nothing unusual in the way proj is used in eckit -- if you can reproduce this (the failure of finding proj) let me know and I'll have a look.